taggit/admin.py

```diff
@@ -14,3 +19,66 @@ class TagAdmin(admin.ModelAdmin):
     ordering = ["name", "slug"]
     search_fields = ["name"]
     prepopulated_fields = {"slug": ["name"]}
+    actions = ["render_tag_form"]
+
+    def get_urls(self):
+        urls = super().get_urls()
+        custom_urls = [
+            path(
+                "merge-tags/",
+                self.admin_site.admin_view(self.merge_tags_view),
+                name="merge_tags",
+            ),
+        ]
+        return custom_urls + urls
+
+    @admin.action(description="Merge selected tags")
+    def render_tag_form(self, request, queryset):
+        selected = request.POST.getlist(admin.helpers.ACTION_CHECKBOX_NAME)
+        if not selected:
+            self.message_user(request, "Please select at least one tag.")
+            return redirect(request.get_full_path())
+
+        selected_tag_ids = ",".join(selected)
+        redirect_url = (
+            f"{request.get_full_path()}merge-tags/?selected_tags={selected_tag_ids}"
+        )
+
+        return redirect(redirect_url)
+
+    def merge_tags_view(self, request):
+        if request.method == "GET":
+            selected_tag_ids = request.GET.get("selected_tags", "").split(",")
+
+            # store selected_tag_ids in session data until they are merged
+            request.session["selected_tag_ids"] = selected_tag_ids
+        else:
+            selected_tag_ids = request.session.get("selected_tag_ids", [])
+
+        if request.method == "POST":
+            form = MergeTagsForm(request.POST)
+            if form.is_valid():
+                new_tag_name = form.cleaned_data["new_tag_name"]
+                new_tag, created = Tag.objects.get_or_create(name=new_tag_name)
+                with transaction.atomic():
+                    for tag_id in selected_tag_ids:
+                        tag = Tag.objects.get(id=tag_id)
+                        tagged_items = TaggedItem.objects.filter(tag=tag)
+                        for tagged_item in tagged_items:
+                            tagged_item.tag = new_tag
+                            tagged_item.save()
```

Comment:
I think that here, we probably should go for an `update` call, even though it won't call `.save`. Something like `TaggedItem.objects.filter(tag=tag).update(tag=new_tag)`. That way we will support tags that already have loads of tagged items.

Reply:
Yeah I think this is cleaner, updating now